projects/media-viewer/src/lib/viewers/image-viewer/image-viewer.component.ts

```diff
@@ -115,10 +122,69 @@ export class ImageViewerComponent implements OnChanges {
   }
 
   public updateAnnotation(updatedAnnotation: Annotation) {
-    const annotations = this.annotationSet.annotations
-      .filter(annotation => annotation.id !== updatedAnnotation.id);
+    const annotations = this.annotationSet.annotations.filter(annotation => annotation.id !== updatedAnnotation.id);
+
     annotations.push(updatedAnnotation);
     this.annotationSet.annotations = annotations;
     this.api.postAnnotationSet(this.annotationSet);
   }
+
+  public onMouseDown(event: MouseEvent) {
+    if (this.annotationSet && this.drawMode) {
+      this.drawStartX = event.pageX - (window.scrollX + this.img.nativeElement.getBoundingClientRect().left);
+      this.drawStartY = event.pageY - (window.scrollY + this.img.nativeElement.getBoundingClientRect().top);
+
+      this.newRectangle.nativeElement.style.display = 'block';
+      this.newRectangle.nativeElement.style.top =  this.drawStartY + 'px';
+      this.newRectangle.nativeElement.style.left = this.drawStartX + 'px';
+
+      event.preventDefault();
+    }
+  }
+
+  public onMouseMove(event: MouseEvent) {
+    if (this.annotationSet && this.drawMode && this.drawStartX > 0 && this.drawStartY > 0) {
+      this.newRectangle.nativeElement.style.height =
+        (event.pageY - this.drawStartY - (window.scrollY + this.img.nativeElement.getBoundingClientRect().top)) + 'px';
+      this.newRectangle.nativeElement.style.width =
+        (event.pageX - this.drawStartX - (window.scrollX + this.img.nativeElement.getBoundingClientRect().left)) + 'px';
+
+      event.preventDefault();
+    }
+  }
+
+  public onMouseUp(event: MouseEvent) {
+    if (this.annotationSet && this.drawMode) {
+      const rectangle = {
+        id: uuid(),
+        x: +this.newRectangle.nativeElement.style.left.slice(0, -2),
+        y: +this.newRectangle.nativeElement.style.top.slice(0, -2),
+        width: +this.newRectangle.nativeElement.style.width.slice(0, -2),
+        height: +this.newRectangle.nativeElement.style.height.slice(0, -2),
+      };
+
+      const annotation = {
+        id: uuid(),
+        annotationSetId: this.annotationSet.id,
+        color: 'FFFF00',
+        comments: [],
+        page: 1,
+        rectangles: [rectangle as Rectangle],
+        type: 'highlight'
+      };
+
+      this.api
+        .postAnnotation(annotation)
+        .subscribe(a => this.annotationSet.annotations.push(a));
+
+      this.drawStartX = -1;
+      this.drawStartY = -1;
+      this.newRectangle.nativeElement.style.display = 'none';
+      this.newRectangle.nativeElement.style.width = '0';
+      this.newRectangle.nativeElement.style.height = '0';
+      this.drawMode.next(false);
+
```

Comment:
this could probably also go to the rectangle component, with the exception of the api call which could be done here

Reply:
same as above